r/R/duckdb.R

```diff
@@ -135,7 +138,7 @@ duckdb_disconnector <- function(con, tbl_name) {
 #'   summarize(mean_mpg = mean(mpg, na.rm = TRUE)) %>%
 #'   to_arrow() %>%
 #'   collect()
-to_arrow <- function(.data, stream = TRUE) {
+to_arrow <- function(.data, as_arrow_query = TRUE) {
```

Comment:
How useful is this argument?

Reply:
Probably (and hopefully) not very, but I wanted to have an escape route in case we see another issue like at the start of this PR where `duckdb::duckdb_fetch_record_batch(res)` fails, but `duckdb::duckdb_fetch_record_batch(res)$read_table()` works (over time, the DuckDB master branch got into a state where _both_ failed consistently, but at the beginning reading the table worked just fine, but accessing the RBR did not. And since we are at the mercy of both of our release cycles for fixing this, the cost of having this escape hatch doesn't seem so bad to me, but I can remove it.

It also helps a bit when debugging / testing (one doesn't have to recreate `to_arrow()` without the wrapper).

Reply:
You can get the RBR from `$.data` in the query object, is that sufficient for debugging purposes?

Reply:
Oh nm, it gets wrapped in a InMemoryDataset. Alright, I don't like this but this is fine, we can prune it later.

Reply:
I could also mark it is a temporary workaround for when we simplify this in the future? It's a little silly to effectively say "this is deprecated" when we introduce it, but I'm not sure when we'll get to doing the simplification + improvements so this can simply emit RBRs and those have the right signaling/methods so that it's clear that they are a good thing to use in dplyr queries